acceptance/tests/direct_puppet/static_catalog_env_control.rb

```diff
@@ -0,0 +1,348 @@
+test_name "Environment control of static catalogs" 
+require 'json'
+
+@testroot = master.tmpdir(File.basename(__FILE__, '/*'))
+### @testroot = "/etc/puppetlabs"
+
+@coderoot = "#{@testroot}/code"
+
+@master_opts = {
+  'main' => {
+    'environmentpath' => "#{@coderoot}/environments",
+  },
+}
+
+@agent_result_dirs = {}
+@catalog_files = {}
+
+agents.each do |agent|
+  @agent_result_dirs[agent.hostname] = agent.tmpdir('results')
+  @catalog_files[agent.hostname] = "#{on(agent, puppet('config', 'print', 'client_datadir')).stdout.chomp}/catalog/#{agent.hostname}.json"
+end
+
+@manifest = <<MANIFEST
+File {
+  ensure => directory,
+  mode => "0755",
+}
+
+file { 
+  '#{@testroot}':;
+  '#{@coderoot}':;
+  '#{@coderoot}/environments':;
+  '#{@coderoot}/environments/production':;
+  '#{@coderoot}/environments/production/manifests':;
+  '#{@coderoot}/environments/production/modules':;
+  '#{@coderoot}/environments/production/modules/hello':;
+  '#{@coderoot}/environments/production/modules/hello/manifests':;
+  '#{@coderoot}/environments/production/modules/hello/files':;
+
+  '#{@coderoot}/environments/canary':;
+  '#{@coderoot}/environments/canary/manifests':;
+  '#{@coderoot}/environments/canary/modules':;
+  '#{@coderoot}/environments/canary/modules/can_hello':;
+  '#{@coderoot}/environments/canary/modules/can_hello/manifests':;
+  '#{@coderoot}/environments/canary/modules/can_hello/files':;
+  
+}
+
+file { '/etc/puppetlabs/puppetserver/conf.d/puppetserver.conf' :
+  ensure => file,
+  mode => "0644",
+  content =>"# configuration for the JRuby interpreters
+jruby-puppet: {
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+    gem-home: /opt/puppetlabs/server/data/puppetserver/jruby-gems
+    master-conf-dir: /etc/puppetlabs/puppet
+    master-code-dir: #{@coderoot}
+    master-var-dir: /opt/puppetlabs/server/data/puppetserver
+    master-run-dir: /var/run/puppetlabs/puppetserver
+    master-log-dir: /var/log/puppetlabs/puppetserver
+}
+http-client: {
+}
+profiler: {
+}
+versioned-code : {
+  code-id-command : #{@coderoot}/code_id.sh
+  code-content-command : #{@coderoot}/code_content.sh
+}
+",
+}
+
+file { '#{@coderoot}/code_id.sh' :
+  ensure => file,
+  mode => "0755",
+  content => '#! /bin/sh
```

Comment:
Use `/bin/bash`, we ran into problems with puppetserver acceptance as some of the master platforms link /bin/sh to dash.